### PR TITLE
Matveibos-hometask-2

### PR DESCRIPTION
**Pull request by Matveibos**
Домашняя работа 2

### CODE REVIEW

File: Matveibos/hometask-2/get_middle_character.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+def run_cli
+  loop do
+    puts 'Enter string'
```
Comment (vol6tmetr) [inline Matveibos/hometask-2/get_middle_character.rb]: It is better to define strings and numbers used in application as a constant variables. You can find more information about this practice if you search "magic strings/numbers"

File: Matveibos/hometask-2/get_middle_character.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    string = gets.chomp
+    break if string == 'exit!'
+
+    if string.empty?
```
Comment (vol6tmetr) [inline Matveibos/hometask-2/get_middle_character.rb]: It is completely fine to use if/else here, but check out how you can rewrite this statement with the ternary expression. You can use them, if it is not affecting readability for the code, and this is the case for this code, IMHO

File: Matveibos/hometask-2/get_middle_character.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      puts 'String can not be blank!'
+    else
+      puts "Middle character: #{get_middle(string)}"
+
```
Comment (vol6tmetr) [inline Matveibos/hometask-2/get_middle_character.rb:9]: Extra empty line (not needed)

File: Matveibos/hometask-2/get_middle_character.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    end
+  end
+end
+
+def get_middle(str)
+  if str.length.even?
```
Comment (vol6tmetr) [inline Matveibos/hometask-2/get_middle_character.rb]: Again, I think that ternary expression will work here as well

File: Matveibos/hometask-2/get_middle_character.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    puts str[str.length / 2 - 1] + str[str.length / 2]
```
Comment (vol6tmetr) [inline Matveibos/hometask-2/get_middle_character.rb]: Check this out, you use `str.length / 2` 3 times here, I think it is better to move it to the variable, if you use this so many times
Also you can use ranges to iterate with the strings in Ruby, check how you can rewrite `get_middle` function with the ranges

File: Matveibos/hometask-2/get_middle_character.rb (hunk 2 above)
Comment (Matveibos) [reply Matveibos/hometask-2/get_middle_character.rb]: Do you mean combine break if string == 'exit!'  and if string.empty? ...  with the ternary expression?

Comment (vol6tmetr) [reply Matveibos/hometask-2/get_middle_character.rb]: No, I mean that you can rewrite that to this:
```
string.empty? ? puts 'String can not be blank!' : puts "Middle character: #{get_middle(string)}"
```

Comment (Matveibos) [reply Matveibos/hometask-2/get_middle_character.rb:11]: I did it like this :
if st.empty? ? (puts 'String can not be blank!') : (puts RESULT + get_middle(st))
# comment, but rubocop warns: Lint/EmptyConditionalBody: Avoid if branches without a body.
end

File: Matveibos/hometask-2/get_middle_character.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+ENTER_STRING = 'enter string'
+ERROR = 'String can not be blank!'
+EXIT = 'exit!'
+RESULT = 'Middle character:'
```
Comment (vol6tmetr) [inline Matveibos/hometask-2/get_middle_character.rb:8]: Add empty line between constants and other parts of code

File: Matveibos/hometask-2/get_middle_character.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  str[slen.floor..slen.ceil]
+end
+
+def run_cli
+  loop do
+    puts ENTER_STRING
+    st = gets.chomp
+    break if st == EXIT
+
+    st.empty? ? (puts 'String can not be blank!') : (puts RESULT + get_middle(st))
```
Comment (vol6tmetr) [inline Matveibos/hometask-2/get_middle_character.rb]: You just moved `'String can not be blank!'` to the `ERROR` constant, so lets use it there
Also prefer string interpolation, rather then concatenation, at least in this case: `RESULT + get_middle(st)`

Comment (Matveibos) [reply Matveibos/hometask-2/get_middle_character.rb]: please excuse my inattention)
Ok!

**Review by vol6tmetr — APPROVED** [review]

**Pull request merged**